front/src/components/boxs/device-in-room/device-features/sensor-value/BadgeNumberDeviceValue.jsx

```diff
@@ -79,15 +80,28 @@ const BADGE_VALUE_CONVERTERS = {
 };
 
 const BadgeNumberDeviceValue = props => {
-  const { category, type, last_value: lastValue = null, unit } = props.deviceFeature;
+  const { user, deviceFeature } = props;
+  const { category, type, last_value: lastValue = null, unit } = deviceFeature;
+
+  const userUnit = user.distance_unit_preference;
 
   const colorMethod = BADGE_CATEGORIES[category];
   if (!colorMethod) {
     return <RawDeviceValue {...props} />;
   }
 
-  let value = lastValue === null ? -1 : lastValue;
   let valueIsEnum = false;
+
+  let value = lastValue;
+  let displayUnit = unit;
+
+  const isDistanceUnit = DISTANCE_UNITS.DEVICE_FEATURE_UNITS.includes(displayUnit);
+  // Conversion if necessary (and if user is present)
+  if (value !== null && isDistanceUnit) {
+    const conversion = convertUnitDistance(lastValue, unit, userUnit);
+    value = conversion.value;
+    displayUnit = conversion.unit;
+  }
   const valued = value !== -1;
```

Comment:
`valued` should be `value === null`, otherwise here any value that's equal to -1 will display "No value"

Reply:
Indeed !!